docs: better validation hook description

baznikin opened the pull request.

jycouet reviewed (changes requested):
Thank you for the PR, we need these to improve the lib.

For `llms*` content, it should update automatically if you start the doc. `cd docs && npm i && npm run dev`
For `ref*` update, it should be done with `npm run build-ref`

Let me know if you want any help 👍

Thx again 🙏

baznikin commented on docs/docs/lifecycle-hooks.md:
```diff
@@ -5,18 +5,41 @@ In Remult, you can take advantage of Entity Lifecycle Hooks to add custom logic
-        throw new Error("Task title must be at least 5 characters long.");
+  @Entity<Order>("orders", {
+    validation: async (order, e) => {
+      if (order.status == "complete" and order.itemsCount == 0) {
```
oops, I am not JS man yet :)

baznikin commented on docs/docs/lifecycle-hooks.md:
```diff
@@ -5,18 +5,41 @@ In Remult, you can take advantage of Entity Lifecycle Hooks to add custom logic
-    validation: async (task, e) => {
-      if (task.title.length < 5) {
-        throw new Error("Task title must be at least 5 characters long.");
+  @Entity<Order>("orders", {
+    validation: async (order, e) => {
+      if (order.status == "complete" and order.itemsCount == 0) {
+        throw new Error("Order without items can't be `complete`!");
       }
     },
   })
```
Can't think out any complex example with task, so picked up another reference with orders/customers. Will fix it

baznikin replied:
BTW, isn't better to set e.fields.xxx.error = 'validation error' instead of throw if we can?

baznikin commented:
I tried to address all comments.
There are changes in other ref docs, I didn't include them in PR (but I can if required)
```
	modified:   docs/docs/ref_entity.md
	modified:   docs/docs/ref_field.md
	modified:   docs/docs/ref_remult.md
	modified:   docs/docs/ref_remultserveroptions.md
```

baznikin commented on docs/docs/ref_entity.md:
```diff
@@ -267,5 +267,17 @@ An arrow function that identifies the `id` column to use for this entity
 Arguments:
 * **ref**
 * **row**
+## dataProvider
```
I added comment to `[/projects/core/src/validators.ts#L110](https://github.com/remult/remult/blob/docs-styles-fixes/projects/core/src/validators.ts#L110)` and to ref_validation.md as well. Other `ref_` files was updated during doc generating process, I suppose changes in source files wasn't reflected in docs. @jycouet added them

baznikin commented:
`llms-*` has conflicts; rebased from `main` and regenerated docs
